internal/connector/internal/handlers/connector_agent.go

```diff
@@ -222,9 +222,34 @@ func (h *ConnectorClusterHandler) presentDeployment(r *http.Request, resource db
 		return private.ConnectorDeployment{}, err
 	}
 
-	apiSpec, err := h.Service.GetConnectorWithBase64Secrets(r.Context(), resource)
+	connectorPhase := resource.Status.Phase
+	// try resetting phase if secrets were previously missing
+	if connectorPhase == dbapi.ConnectorStatusPhaseSecretsUnavailable {
+		connectorPhase = dbapi.ConnectorStatusPhaseAssigned
+	}
+
+	// avoid ignoring this deployment altogether if there is an issue in getting secrets from the vault
+	ctx := r.Context()
+	apiSpec, invalidSecrets, err := h.Service.GetConnectorWithBase64Secrets(ctx, resource)
 	if err != nil {
-		return private.ConnectorDeployment{}, err
+		if invalidSecrets {
+			// log error in getting secrets and signal that connector spec doesn't have secrets
+			glog.Errorf("Error getting connector %s with base64 secrets: %s", apiSpec.ID, err)
+
+			apiSpec.ConnectorSpec = []byte("{}")
+			connectorPhase = dbapi.ConnectorStatusPhaseSecretsUnavailable
+		} else {
+			return private.ConnectorDeployment{}, err
+		}
+	}
+
+	// check for phase change/reset due to secrets handling
+	if connectorPhase != resource.Status.Phase {
+		converted.Status.Phase = private.ConnectorState(connectorPhase)
+		resource.Status.Phase = connectorPhase
+		if err := h.Service.UpdateConnectorDeploymentStatus(ctx, resource.Status); err != nil {
```

Comment:
Phase updates to `secrets-missing` and reset back to `assigned` are persisted in the DB. Users will be able to see it in the UI.

Reply:
Removed fleetmanager DB updates.